backend/civic_intelligence.py

```
@@ -0,0 +1,246 @@
+            "grievance_count": len(recent_grievances)
+        }
+
+        self._save_snapshot(snapshot)
+
+        # 7. Persist updated weights
+        self.weights.save_weights()
+
+        logger.info(f"Daily refinement complete. Index: {index_score['score']}")
+        return snapshot
+
+    def _optimize_weights(self, issues: List[Issue], grievances: List[Grievance]) -> List[str]:
```

Ruff (0.15.1)

[warning] Unused method argument: `issues`

(ARG002)